src/evt/cam_api.c

```diff
@@ -545,9 +545,88 @@ ApiStatus WaitForCam(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "evt/cam_api", SetCamProperties, Evt* script, s32 isInitialCall);
+ApiStatus SetCamProperties(Evt* script, s32 isInitialCall) {
+    f32 hitX, hitY, hitZ, hitDepth, nX, nY, nZ;
+    s32 zoneID;
+    Bytecode* args = script->ptrReadPos;
+    Camera* camera = &gCameras[evt_get_variable(script, *args++)];
+    f32 moveSpeed = evt_get_float_variable(script, *args++);
+    f32 posX = evt_get_float_variable(script, *args++);
+    f32 posY = evt_get_float_variable(script, *args++);
+    f32 posZ = evt_get_float_variable(script, *args++);
+    f32 boomLength = evt_get_float_variable(script, *args++);
+    f32 boomPitch = evt_get_float_variable(script, *args++);
+    f32 viewPitch = evt_get_float_variable(script, *args++);
 
-INCLUDE_ASM(s32, "evt/cam_api", AdjustCam, Evt* script, s32 isInitialCall);
+    if (isInitialCall) {
+        hitDepth = 32767.0f;
+        zoneID = test_ray_zones(posX, posY + 10, posZ, 0, -1.0f, 0, &hitX, &hitY, &hitZ, &hitDepth, &nX, &nY, &nZ);
+        if (zoneID >= 0) {
+            camera->controlSettings = *gZoneCollisionData.colliderList[zoneID].camSettings;
+        }
+
+        camera->movePos.x = posX;
+        camera->movePos.y = posY;
+        camera->movePos.z = posZ;
+        camera->controlSettings.boomLength = boomLength;
+        camera->controlSettings.boomPitch = boomPitch;
+        camera->controlSettings.viewPitch = viewPitch;
+        camera->moveSpeed = moveSpeed;
+        camera->unk_506 = 1;
+        camera->followPlayer = TRUE;
+        camera->panPhase = 0;
+        return ApiStatus_BLOCK;
+    }
+
+    if (camera->interpAlpha >= 1) {
```

Comment:
`> 0` might make more sense here (not sure if it'll optimize the same though)

Reply:
interpAlpha is float. Should I put 1.0f instead of 1?

Reply:
ohhh then that will not work, right. yes 1.0f please :)